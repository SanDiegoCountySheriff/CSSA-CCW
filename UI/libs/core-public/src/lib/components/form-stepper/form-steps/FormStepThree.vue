<template>
  <div>
    <v-form
      ref="form"
      v-model="valid"
    >
      <v-row class="ml-5">
        <v-col
          cols="6"
          md="5"
          sm="3"
        >
          <TextInput
            :label="'Address line 1'"
            :rules="[ v => !!v || 'Address line 1 cannot be blank']"
            :target="'addressLine1'"
            @input="(v,t) => {handleInput(v,t)}"
          />
        </v-col>

        <v-col
          cols="6"
          md="5"
          sm="3"
        >
          <TextInput
            :label="'Address line 2'"
            :target="'addressLine2'"
            @input="(v,t) => {handleInput(v,t)}"
          />
        </v-col>
      </v-row>

      <v-row class="ml-5">
        <v-col
          cols="6"
          md="5"
          sm="3"
        >
          <TextInput
            :label="'City'"
            :target="'city'"
            :rules="[ v => !!v || ' City cannot be blank']"
            @input="(v,t) => {handleInput(v,t)}"
          />
        </v-col>

        <v-col
          cols="6"
          md="5"
          sm="3"
        >
          <TextInput
            :label="'State'"
            :target="'state'"
            :rules="[ v => !!v || 'State cannot be blank']"
            @input="(v,t) => {handleInput(v,t)}"
          />
        </v-col>
<v-col
          cols="6"
          md="5"
          sm="3"
        >
          <TextInput
            :label="'County'"
            :target="'county'"
            :rules="[ v => !!v || 'County cannot be blank']"
            @input="(v,t) => {handleInput(v,t)}"
          />
        </v-col>
        <v-col
          cols="6"
          md="5"
          sm="3"
        >
          <TextInput
            :label="'Zip'"
            :target="'zip'"
            :rules="[ v => !!v || 'Zip cannot be blank']"
            @input="(v,t) => {handleInput(v,t)}"
          />
        </v-col>

        <v-col
          cols="6"
          md="5"
          sm="3"
        >
          <v-text-field
            v-model="address.country"
            label="Country"
            :rules="[ v => !!v || 'Country cannot be blank']"
          />
        </v-col>
      </v-row>

      <v-row>
        <PreviousAddressDialog :get-previous-address-from-dialog="getPreviousAddressFromDialog" />
      </v-row>
    </v-form>
  </div>
</template>

<script lang='ts'>

import { AddressInfo } from '@shared-ui/types/defualtTypes';
import { defineComponent, PropType } from 'vue';
import PreviousAddressDialog from '@shared-ui/components/dialogs/PreviousAddressDialog.vue';
import TextInput from '@shared-ui/components/inputs/TextInput.vue';

export default defineComponent( {
  name: 'FormStepThree',
  components: { TextInput, PreviousAddressDialog },
  props: {
    handleNextSection: {
      type: Function as PropType<() => void>,
      default: () => null,
    },
  },
  data(){
    return {
      address: {} as AddressInfo,
      previousAddress: [] as Array<AddressInfo>,
      valid: false
    }
  },
  methods:{
     getPreviousAddressFromDialog(address: AddressInfo){
       this.previousAddress.push(address)
     },
    handleInput(value, target){
        switch(target){
          case "addressLine1":
            this.address.addressLine1 = value
            break
          case "addressLine2":
            this.address.addressLine2 = value
            break
          case "city":
            this.address.city = value
            break
          case "state":
            this.address.state = value
            break
          case "zip":
            this.address.zip = value
            break
          case "county":
            this.address.county = value
            break
          case "country":
            this.address.country = value
            break
          default:
            return
        }
    }
  }
});
</script>

<style lang='scss' scoped>
.subHeader {
  font-size: 1.5rem;
}
</style>
