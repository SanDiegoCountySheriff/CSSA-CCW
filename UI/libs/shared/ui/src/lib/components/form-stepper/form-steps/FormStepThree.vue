<template>
  <div>
    <v-form
      ref='form'
      v-model='valid'
    >
      <v-row class='ml-5'>
        <v-col
          cols='6'
          md='5'
          sm='3'
          >
          <v-text-field
            v-model='address.addressLine1'
            label='Address line 1'
            :rules='[ v => !!v || "Address line 1 cannot be blank"]'
           />
        </v-col>

        <v-col
          cols='6'
          md='5'
          sm='3'
          >
          <v-text-field
            v-model='address.addressLine2'
            label='Address line 2'
           />
        </v-col>
      </v-row>

      <v-row class='ml-5'>
          <v-col
          cols='6'
          md='5'
          sm='3'
          >
          <v-text-field
            v-model='address.city'
            label='City'
            :rules='[ v => !!v || " City cannot be blank"]'
           />
        </v-col>

          <v-col
          cols='6'
          md='5'
          sm='3'
          >
          <v-text-field
            v-model='address.state'
            label='State'
            :rules='[ v => !!v || "State cannot be blank"]'
           />
        </v-col>

          <v-col
          cols='6'
          md='5'
          sm='3'
          >
          <v-text-field
            v-model='address.zip'
            label='Zip'
            :rules='[ v => !!v || "Zip cannot be blank"]'
           />
        </v-col>

        <v-col
          cols='6'
          md='5'
          sm='3'
        >
          <v-text-field
            v-model='address.country'
            label='Country'
            :rules='[ v => !!v || "Country cannot be blank"]'
          />
        </v-col>
      </v-row>

     <v-row>
      <PreviousAddressDialog :getPreviousAddressFromDialog="getPreviousAddressFromDialog"/>
     </v-row>

    </v-form>
  </div>
</template>

<script lang='ts'>

import { AddressInfo } from '@shared-ui/types/defualtTypes';
import { defineComponent, PropType } from 'vue';
import PreviousAddressDialog from '@shared-ui/components/dialogs/PreviousAddressDialog.vue';

export default defineComponent( {
  name: 'FormStepThree',
  components: { PreviousAddressDialog },
  props: {
    handleNextSection: {
      type: Function as PropType<() => void>,
      default: () => null,
    },
  },
  data(){
    return {
      address: {} as AddressInfo,
      previousAddress: [] as Array<AddressInfo>,
      valid: false
    }
  },
  methods:{
     getPreviousAddressFromDialog(address: AddressInfo){
       this.previousAddress.push(address)
     }
  }
});
</script>

<style lang='scss' scoped>
.subHeader {
  font-size: 1.5rem;
}
</style>
