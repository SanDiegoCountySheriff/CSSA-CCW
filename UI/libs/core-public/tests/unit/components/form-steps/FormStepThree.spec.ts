import FormStepThree from '@core-public/components/form-stepper/form-steps/FormStepThree.vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';

const pinia = createTestingPinia();
const localVue = createLocalVue();
const tMock = {
  $t: text => text,
};

describe('FormStepThree', () => {
  let vuetify;
  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify();
    wrapper = mount(FormStepThree, {
      localVue,
      vuetify,
      pinia,
      mocks: tMock,
      propsData: {
        handleNextSection: () => null,
      },
    });
  });

  it('Should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Should render all the inital text inputs', () => {
    expect(wrapper.findAllComponents('.v-text-field').length).toBe(7);
  });

  it('Should render the next text inputs', async () => {
    await wrapper.find('#different-mailing').setChecked();
    expect(wrapper.findAllComponents('.v-text-field').length).toBe(14);
  });
});
