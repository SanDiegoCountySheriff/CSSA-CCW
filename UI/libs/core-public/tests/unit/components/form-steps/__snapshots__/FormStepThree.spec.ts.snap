// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormStepThree Should match the snapshot 1`] = `
<div>
  <div class="v-subheader sub-header font-weight-bold theme--light">
    Address Information
  </div>
  <form novalidate="novalidate" class="v-form">
    <div class="row ml-5">
      <div class="col-sm-3 col-md-5 col-6">
        <div class="v-input theme--light v-text-field">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-4" class="v-label theme--light" style="left: 0px; position: absolute;">Address line 1</label><input id="input-4" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-sm-3 col-md-5 col-6">
        <div class="v-input theme--light v-text-field">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-7" class="v-label theme--light" style="left: 0px; position: absolute;">Address line 2</label><input id="input-7" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="row ml-5">
      <div class="col-sm-3 col-md-5 col-6">
        <div class="v-input theme--light v-text-field">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-10" class="v-label theme--light" style="left: 0px; position: absolute;">City</label><input id="input-10" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-sm-3 col-md-5 col-6">
        <div class="v-input theme--light v-text-field">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-13" class="v-label theme--light" style="left: 0px; position: absolute;">State</label><input id="input-13" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-sm-3 col-md-5 col-6">
        <div class="v-input theme--light v-text-field">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-16" class="v-label theme--light" style="left: 0px; position: absolute;">County</label><input id="input-16" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-sm-3 col-md-5 col-6">
        <div class="v-input theme--light v-text-field">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-19" class="v-label theme--light" style="left: 0px; position: absolute;">Zip</label><input id="input-19" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-sm-3 col-md-5 col-6">
        <div class="v-input theme--light v-text-field">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-22" class="v-label theme--light" style="left: 0px; position: absolute;">Country</label><input id="input-22" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    <div class="row ml-5 my-5">
      <div class="v-input theme--light v-input--selection-controls v-input--checkbox">
        <div class="v-input__control">
          <div class="v-input__slot">
            <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="different-mailing" role="checkbox" type="checkbox" value="">
              <div class="v-input--selection-controls__ripple"></div>
            </div><label for="different-mailing" class="v-label theme--light" style="left: 0px; position: relative;">Different Mailing address</label>
          </div>
          <div class="v-messages theme--light">
            <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
          </div>
        </div>
      </div>
    </div>
    <!---->
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    <div class="v-subheader sub-header font-weight-bold theme--light">
      Previous Address
    </div>
    <div class="previous-address-container">
      <div class="container container--fluid">
        <div class="v-data-table my-3 theme--light">
          <div class="v-data-table__wrapper">
            <table>
              <thead>
                <tr>
                  <th> Address line 1</th>
                  <th>Address line 2</th>
                  <th>City</th>
                  <th>State</th>
                  <th>County</th>
                  <th>Zip</th>
                  <th>Country</th>
                </tr>
              </thead>
              <tbody></tbody>
            </table>
          </div>
        </div>
      </div>
      <div><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary my-5" id="add-previous-address-btn" role="button" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content">
         Add previous address
      </span></button>
        <div class="v-dialog__container">
          <!---->
        </div>
      </div>
    </div>
    <div class="form-btn-container"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
    Continue
  </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default secondary mx-2"><span class="v-btn__content">
    Save and Exit
  </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default error mr-2"><span class="v-btn__content">
    Cancel
  </span></button></div>
  </form>
</div>
`;
